features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -2902,6 +2902,18 @@
     <!--This is the separator that use to separate multiple roles in the role claim value coming from IDP side-->
     <FederatedIDPRoleClaimValueAttributeSeparator>{{federated.idp.role_claim_value_attribute_separator}}</FederatedIDPRoleClaimValueAttributeSeparator>
 
+     <!--This enables federated user's role based authorization -->
+     <EnableFederatedIDPRoleBasedAuthorization>{{federated.idp.role_based_authorization}}</EnableFederatedIDPRoleBasedAuthorization>
```

Comment:
the config name need to be revisited.
[federated] as top level config doesn't make sense

if this will be used only for system apps, better put this under [system_applications]

Reply:
If we generalize this solution for product-is, we can use this flow for all the applications (not bounded to system applications) & it will be more like improvement to federated login flow (regardless of application type), WDYT?

Reply:
Going with `system_applications` top level